interactors/nonceHandlerV3/addressNonceHandler.go

```diff
@@ -92,26 +108,37 @@ func (anh *addressNonceHandler) Close() {
 	anh.cancelFunc()
 }
 
-func (anh *addressNonceHandler) fetchGasPriceIfRequired(ctx context.Context) {
+func (anh *addressNonceHandler) applyGasPriceIfRequired(ctx context.Context, tx *transaction.FrontendTransaction) {
+	anh.mut.Lock()
```

Comment:
why this change? This will keep the lock as long at the API get call is done.

Reply:
changed.